cmd/fossa/report.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	logging "github.com/op/go-logging"
+	"github.com/urfave/cli"
+
+	"github.com/fossas/fossa-cli/config"
+)
+
+var reportLogger = logging.MustGetLogger("report")
+
+type dependencyResponse struct {
+	Loc struct {
+		Package  string
+		Revision string
+	}
+	Licenses []licenseResponse
+	Project  struct {
+		Title   string
+		URL     string
+		Authors []string
+	}
+}
+
+type licenseResponse struct {
+	ID       string
+	Title    string
+	FullText string
+}
+
+func reportCmd(c *cli.Context) {
+	conf, err := config.New(c)
+	if err != nil {
+		reportLogger.Fatalf("Could not load configuration: %s", err.Error())
+	}
+
+	server, err := url.Parse(conf.Endpoint)
+	if err != nil {
+		reportLogger.Fatalf("Invalid FOSSA endpoint: %s", err.Error())
+	}
+	api, err := server.Parse(fmt.Sprintf("/api/revisions/%s/dependencies", url.PathEscape(config.MakeLocator(conf.Project, conf.Revision))))
```

Comment:
Why is `MakeLocator` in the `config` package?

Reply:
Legacy. I'll refactor it out later, I think.

Reply:
It kinda makes sense as a config util, to take config params and make it into a certain form.